### PR TITLE
[Dy2static] fix some print transformer problems

### PR DESCRIPTION
**Pull request by chenwhql**
该PR修复三个Print Transformer的问题：

1. 动转静`print`替换为`fluid.layers.Print`之后，如果是gast.Expr而不是gast.Assign，print将会成为叶子节点，这导致的在exe.run(use_prune=True)时，Print节点会被剪掉

- 因此改为，将`print(x)`要处理为`x = fluid.layers.Print(x)`
  - 对于py3：两种情况
    - Expr(Call) - > Assign()
    - Assign() - > Assign()
  - 对于py2：
    - Print() -> Assign()

2. 改为`layers.Print`之后，一些参数不能仅使用默认的，否则和原动态图写法语义存在差别

- `print_phase`应该只包含前向：在动态图的语义中，`print(x)`是只打印当前变量的，所以转换之后，也应该保持语义相同，带参数`print_phase='forward'`

- `summarize`应该改为-1，打印全部tensor数据

3. 一些报错信息应该去掉，换成WARNING，无法转换的并不应该报错

related PR：https://github.com/PaddlePaddle/Paddle/pull/24068

### CODE REVIEW

File: python/paddle/fluid/dygraph/dygraph_to_static/print_transformer.py [hunk 1]
```diff
@@ -39,45 +45,94 @@ def transform(self):
...
+        return None
+
+    def _need_transform(self, var_node, print_node):
+        if isinstance(var_node, gast.Name):
+            if self._is_tensor_node(var_node):
+                return True
             else:
-                raise TypeError(
-                    "print object type error, only support print Variable now.")
+                _logger.warning(
```
Comment (chenwhql) [inline python/paddle/fluid/dygraph/dygraph_to_static/print_transformer.py:123]: 试了下`warnings.wain`，会把原来的code也打出来，有点乱，所以换了logging

**Review by Aurelius84 — DISMISSED** [review]
LGTM

**Review by liym27 — DISMISSED** [review]
LGTM

File: python/paddle/fluid/dygraph/dygraph_to_static/print_transformer.py [hunk 2]
```diff
@@ -39,45 +45,94 @@ def transform(self):
         self.visit(self.root)
 
     # NOTE: deal with print in PY3
-    def visit_Call(self, node):
-        assert isinstance(node, gast.Call)
-        if isinstance(node.func, gast.Name) and node.func.id == 'print':
-            var = self._get_print_var(node)
-            return self._construct_print_node(var)
+    def visit_Assign(self, node):
```
Comment (zhhsplendid) [inline python/paddle/fluid/dygraph/dygraph_to_static/print_transformer.py]: This code contains bug, for example:
If print gast.Call node is not current node's value. Like:

`b = 1 or print("3")`

python is allowed to write above code....

File: python/paddle/fluid/dygraph/dygraph_to_static/print_transformer.py [hunk 3]
```diff
@@ -39,45 +45,94 @@ def transform(self):
...
+        if isinstance(node.value, gast.Call):
+            if isinstance(node.value.func,
+                          gast.Name) and node.value.func.id == 'print':
+                node.value = self._transform_call_node(node.value)
+        return node
+
+    def visit_Expr(self, node):
```
Comment (zhhsplendid) [inline python/paddle/fluid/dygraph/dygraph_to_static/print_transformer.py]: Same for bugs like visit_Assign above. For example, an Expr can be 

`a1 and a2 and a3 and print(x)`

BTW, why not write the visit_Call like the old code? If you are handling the `x = print(a)` case, you could get parent node of Call node from static analysis and then handle that. In my opinion it is better than current code.

File: python/paddle/fluid/dygraph/dygraph_to_static/print_transformer.py (hunk 2 above)
Comment (chenwhql) [reply python/paddle/fluid/dygraph/dygraph_to_static/print_transformer.py]: done, thx. for this case, keep `print` is `gast.Call `, do not replace by `gast.Assign`

File: python/paddle/fluid/dygraph/dygraph_to_static/print_transformer.py (hunk 3 above)
Comment (chenwhql) [reply python/paddle/fluid/dygraph/dygraph_to_static/print_transformer.py]: done, thx. for this case, keep `print` is `gast.Call`, do not replace by `gast.Assign`

using static analysis is efficient!

**Review by zhhsplendid — APPROVED** [review]
LGTM

**Pull request merged**